sdk/src/types/block/block_id.rs

```diff
@@ -1,11 +1,104 @@
 // Copyright 2020-2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-impl_id!(
-    pub BlockId,
-    32,
-    "A block identifier, the BLAKE2b-256 hash of the block bytes. See <https://www.blake2.net/> for more information."
-);
+use super::{slot::SlotIndex, ConvertTo};
+use crate::types::block::Error;
 
+impl_id!(pub BlockHash, 32, "The hash of a [`Block`].");
```

Comment:
In the context of the BlockId how it's used now, this is called `ID` in the TIP
`ID` is weird, but `BlockHash` is also used in the context of signature with this definition:
```
`Block Hash` be the BLAKE2b-256 hash over the serialized content of the _Block_ field.
````
So I think for this hash here we should come up with a different name

Reply:
I dunno, it feel appropriate

Reply:
How about BlockWrapperHash? Then `BlockHash` can be returned from `block_hash()` instead of `[u8; 32]`

Reply:
But the type isn't useful there, we would always just consume it

Reply:
Ok, but even if the type isn't needed there, from the name it would alone one could expect that it's the one there
But it's a bit of a mess with the naming of the blocks anyways